go/vt/vtgate/semantics/early_rewriter.go

```diff
@@ -68,8 +68,31 @@ func (r *earlyRewriter) down(cursor *sqlparser.Cursor) error {
 		}
 	case *sqlparser.Order:
 		r.clause = "order clause"
-	case sqlparser.GroupBy:
+	case *sqlparser.GroupBy:
 		r.clause = "group statement"
+		if node.All {
+			// Get the current scope and find the select statement from it
+			selStmt, isSelect := r.scoper.currentScope().stmt.(*sqlparser.Select)
+			if !isSelect {
+				return vterrors.Errorf(vtrpcpb.Code_INTERNAL, "found a group but the current scope is not a select statement")
```

Comment:
how about we do this in the normalizer? And from there own we will only have []Expr in the groupBy.

Reply:
seems reasonable

Reply:
I am going ahead with the changes to move this logic to the normaliser, but just so that all of us are on the same page, it would mean that this syntactic sugar will only work if the vtgate is configured to have normalisation turned on.

Reply:
We have the `MustRewriteAST` function that we can change too, to make the normalizer run always if the `group by all` syntax is used, but I don't think we should do that

Reply:
Doing this change in the normaliser has an added advantage of also working with v3, 😆

Reply:
Done